### PR TITLE
W13-YlvaK - Express API

### PR DESCRIPTION
**Pull request by YlvaKarlsson**
Simple frontend:
https://w13-frontend-express-api-ylva.netlify.app/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -18,11 +11,37 @@ const app = express();
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
+app.use(express.urlencoded({extended:false}));
```
Comment (EmmaHoltegaard) [inline server.js:14]: This is new to me, I think. What does this do?

File: server.js [hunk 2]
```diff
@@ -18,11 +11,37 @@ const app = express();
...
-});
+app.get("/books", (req, res) => {
+  res.send(booksData);
+})
+// added error-handlers for 404-responses
+// because the isbn is a number and the parameter is a string - I needed to add the toString-function after the books isbn-number 
+// One endpoint to return a single result (one book), based on the isbn-number requested in the url
+app.get("/books/isbn/:isbn", (req, res) => {
+  let hits = booksData.filter(book => book.isbn.toString() === req.params.isbn)
+  if (hits.length === 0) {res.status(404).send("Sorry, no book with that ISBN-nr exists =(")}
```
Comment (EmmaHoltegaard) [inline server.js:25]: Good job adding error messages to your API.

File: server.js [hunk 3]
```diff
@@ -18,11 +11,37 @@ const app = express();
...
+  res.send(hits[0])
+})
+// query parameter - filters on author or title using query parameters
+app.get("/books/authors", (req, res) => {
+  let author = req.query.author
+  let title = req.query.title
+  let hits = booksData
+  if (author !== undefined) {hits = hits.filter(book => book.authors === author)}
```
Comment (EmmaHoltegaard) [inline server.js:33]: Didn't know you could used 'undefined' this way. Cool!

**Review by EmmaHoltegaard — COMMENTED** [review]
Good job, Ylva (and sorry for doing my review so last minute!). You completed all the requirements and managed some stretch goals as well. I really like that you added error messages to the API, even though it wasn't part of the "basic" materials/requirements this week- and you added them in a slightly different way than I've seen other places. Cool!

**Review by SandraMadeleine — COMMENTED** [review]
Was initially a bit confused since your code was already reviewed, but I saw that Emma also reviewed Nina. :-D Your code looks great Ylva and you also managed to connect a frontend, super good job! I really like the way you have both written and structured your code, it looks very professional!

**Pull request closed without merging**